Add Support for Toshiba TMPM4G9

ganesh-ramachandran opened the pull request:
### Description

<!-- 
    Required
    Add here detailed changes summary, testing results, dependencies 
    Good example: https://os.mbed.com/docs/latest/reference/workflow.html (Pull request template)
-->
Add mbed support for TOSHIBA's TMPM4G9 board.

Tested with all toolchains and test results are attached.
[M4G9_GreenteaAutomated_TestResults.txt](https://github.com/ARMmbed/mbed-os/files/2328024/M4G9_GreenteaAutomated_TestResults.txt) 
[M4G9_SingleAutomated_TestResults.txt]( https://github.com/ARMmbed/mbed-os/files/2328027/M4G9_SingleAutomated_TestResults.txt)


### Pull request type

<!-- 
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [ ] Fix
    [ ] Refactor
    [x] New target
    [ ] Feature
    [ ] Breaking change

Commit:
Added Support for TOSHIBA TMPM4G9

ghost commented:
Rebased

ghost commented:
Hi @0xc0170, any status update on this PR.

0xc0170 commented:
> Hi @0xc0170, any status update on this PR.

I'll review it today

0xc0170 commented on targets/TARGET_TOSHIBA/TARGET_TMPM4G9/us_ticker.c:
```diff
@@ -0,0 +1,102 @@
+    NVIC_ClearPendingIRQ(INTT32A00_A_CT_IRQn);
+}
+
+void us_ticker_free(void)
```
why this function is empty?

0xc0170 commented on targets/TARGET_TOSHIBA/TARGET_TMPM4G9/device/TOOLCHAIN_IAR/tmpm4g9f15.icf:
```diff
@@ -0,0 +1,45 @@
+define symbol __ICFEDIT_region_BRAM_end__    = 0x200307FF;
+/*-Sizes-*/
+/*Heap 1/4 of ram and stack 1/8*/
+define symbol __ICFEDIT_size_cstack__   = 0xC000;
```
1k stack size

Commit:
Implemented us_ticker free() & modified IAR stack

cmonr commented:
/morph build

0xc0170 commented:
@ganesh-ramachandran Can you review the 2 failures ? What IAR version is required? This requires a new cmsis pack, will request the update (uvision failure)

ghost commented:
> Can you review the 2 failures ? What IAR version is required? This requires a new cmsis pack, will request the update (uvision failure)

@0xc0170, for IAR - issue is due to IAR version. Issue occurs in IAR V7.5... We are working with IAR 7.8... and no issues with that. For uvision failure - M4G9 keil pack to be installed (http://www.keil.com/dd2/toshiba/tmpm4g9f15fg/)

0xc0170 commented:
The pack update was requested to be added (will be in couple of days). The IAR should be 7.8 for exporters, I noticed the build utility reportes 7.5.

@OPpuolitaival @studavekar What is the IAR version used by exporters?

0xc0170 commented:
@ganesh-ramachandran Can you rebase? I would like to restart testing (we updated cmsis packs for uvision, need to check if this MCU is also in or will need to request one).

Sorry for the delay, there were various CI challenges in the last 2 weeks

ghost commented:
> @ganesh-ramachandran Can you rebase? I would like to restart testing (we updated cmsis packs for uvision, need to check if this MCU is also in or will need to request one).
> 
> Sorry for the delay, there were various CI challenges in the last 2 weeks

@0xc0170, Rebased. Thank you for the confirmation on cmsis packs for uvision. IAR exporter version is also fixed?

0xc0170 commented:
/morph export-build

cmonr commented:
@ganesh-ramachandran Could you refer us to the IAR 7.x release notes where the device was added?

We were able to find this, but it's for 8.30: http://supp.iar.com/FilesPublic/UPDINFO/013240/arm/doc/infocenter/readme.ENU.html

ghost commented:
> @ganesh-ramachandran Could you refer us to the IAR 7.x release notes where the device was added?
> 
> We were able to find this, but it's for 8.30: 

@cmonr Yes, TMPM4G9 is added in IAR 8.30, We have compiled and tested using mbed-cli with IAR 7.8. May be exporter build might succeed with IAR 7.8. Or when IAR 8.30 would be effective.

adbridge commented:
/morph build

0xc0170 commented:
> @cmonr Yes, TMPM4G9 is added in IAR 8.30, We have compiled and tested using mbed-cli with IAR 7.8. May be exporter build might succeed with IAR 7.8. Or when IAR 8.30 would be effective.

Based on this, exporters should be disabled for now (not supported in 7.x version), please remove iar definition.

We requested cmsis pack update in the CI to get exporters green

cmonr commented:
Whoops. Jumped the gun again on restarting the export job.

Iar network license issue with mbed2 build
 /morph mbed2-build

Commit:
Removed IAR definition for TMPM4G9

TMPM4G9 device was added in IAR 8.30. Based on this, exporters should be disabled for now (not supported in 7.x version).

ghost commented:
> Based on this, exporters should be disabled for now (not supported in 7.x version), please remove iar definition.
> 
> We requested cmsis pack update in the CI to get exporters green

@0xc0170 Removed IAR definition. Requesting to start CI tests again.

NirSonnenschein commented:
/morph build

NirSonnenschein commented:
failures seem to be iar license related:
/morph export-build
/morph mbed2-build

ghost commented:
> d like to restart testing (we updated cmsis packs for uvision, need to check if this MCU is also in or will need to request one).

cmsis pack for uvision to be updated to add this MCU (https://www.keil.com/dd2/toshiba/tmpm4g9f15fg/). 
Requesting to update this pack to fix the uvision exporter build failure for TMPM4G9.

cmonr commented:
@ganesh-ramachandran Correct. We still have an internal ticket open to add it into our expoter VM seeds.

cmonr commented:
Should be ready now!

/morph export-build
